pkg/tcpip/link/fdbased/endpoint.go

```diff
@@ -196,8 +196,12 @@ type Options struct {
 // option for an FD with a fanoutID already in use by another FD for a different
 // NIC will return an EINVAL.
 //
+// Since fanoutID must be unique within the network namespace, we start with
+// the PID to avoid collisions. The only way to be sure of avoiding collisions
+// is to run in a new network namespace.
+//
 // Must be accessed using atomic operations.
-var fanoutID int32 = 0
+var fanoutID int32 = int32(unix.Getpid())
```

Comment:
>   A socket  joins a fanout group by calling setsockopt(2) with level
>               SOL_PACKET and option PACKET_FANOUT.  Each network
>               namespace can have up to 65536 independent groups.  A
>               socket selects a group by encoding the ID in the first 16
>               bits of the integer option value.  

I think that you have to cast this to a uint16 integer, or it can go out of limits
https://github.com/google/gvisor/pull/6128/files#diff-5f5e8994b5677b1b67ef377306819b9ef5d9fa35b4f5e569f55f68f85049084dR301

Reply:
yeah, right, I come up with the wrong solution, I wanted to mean, it has to be a value between 0 and 65536 XD

Reply:
hmm, but that conversion is after the check of the value
https://github.com/google/gvisor/pull/6128/files#diff-5f5e8994b5677b1b67ef377306819b9ef5d9fa35b4f5e569f55f68f85049084dR300-R303
I think that this way, it will fail before you truncate it

Reply:
You're right, I'll fix it.